src/hooks/useTitle.ts

```diff
@@ -6,206 +6,183 @@
+    defaultMessage: `Extensions > {extensionId, select,
+      VotingReputation {Governance}
+      OneTxPayment {One Transaction Payment}
+      other {{extensionId}}
+      } > Setup | Colony - {colonyName}`,
+  },
+
+  colonyMembers: {
+    id: `${displayName}.useTitle.colonyMembers`,
+    defaultMessage: `Members | Colony - {colonyName}`,
+  },
+
+  userProfile: {
+    id: `${displayName}.useTitle.userProfile`,
+    defaultMessage: `Users > {username} | Colony`,
+  },
+
+  transactionDetails: {
+    id: `${displayName}.useTitle.transactionDetails`,
+    defaultMessage: `Transaction - {transactionHash} | Colony - {colonyName}`,
+  },
+  buyTokens: {
+    id: `${displayName}.useTitle.buyTokens`,
+    defaultMessage: `Buy Tokens | Colony - {colonyName}`,
+  },
+  fallbackTitle: {
+    id: `${displayName}.useTitle.fallbackTitle`,
+    defaultMessage: `Colony`,
+  },
+});
+
+interface MessageWithValues {
+  msg: MessageDescriptor;
+  values?: SimpleMessageValues;
+}
+
+const COLONY_HOME_ROUTE_WITHOUT_WILDCARD = COLONY_HOME_ROUTE.slice(0, -2);
+
+const routeMessages: Record<string, MessageDescriptor> = {
+  [CREATE_COLONY_ROUTE]: MSG.createColony,
+  [CREATE_USER_ROUTE]: MSG.createUser,
+  [USER_EDIT_ROUTE]: MSG.editProfile,
+  [NOT_FOUND_ROUTE]: MSG.notFound,
+  [LANDING_PAGE_ROUTE]: MSG.landing,
+  [COLONY_HOME_ROUTE]: MSG.colonyHome,
+  [COLONY_EVENTS_ROUTE]: MSG.colonyEvents,
+  [COLONY_FUNDING_ROUTE]: MSG.colonyFunds,
+  [`${COLONY_HOME_ROUTE_WITHOUT_WILDCARD}${COLONY_EXTENSIONS_ROUTE}`]:
+    MSG.colonyExtensions,
+  [`${COLONY_HOME_ROUTE_WITHOUT_WILDCARD}${COLONY_EXTENSION_DETAILS_ROUTE}`]:
+    MSG.colonyExtensionDetails,
+  [`${COLONY_HOME_ROUTE_WITHOUT_WILDCARD}${COLONY_EXTENSION_SETUP_ROUTE}`]:
+    MSG.colonyExtensionSetup,
+  [USER_ROUTE]: MSG.userProfile,
+  [ACTIONS_PAGE_ROUTE]: MSG.transactionDetails,
+  [COLONY_MEMBERS_ROUTE]: MSG.colonyMembers,
+  '/': MSG.fallbackTitle,
+};
+
+const allRoutes = Object.keys(routeMessages);
+
+const getMessageAndValues = (locationPath: string): MessageWithValues => {
+  const filteredRoutes = allRoutes.filter((routePattern) =>
+    notNull(matchPath(routePattern, locationPath)),
+  );
+
+  // Fallback when no route matches
+  // For example before an invalid route get redirected to 404
+  const matchedRoute = filteredRoutes.at(-1) || '/';
+
+  const values = matchPath(matchedRoute, locationPath)?.params; // this can be empty {}
+
+  return { msg: routeMessages[matchedRoute], values };
+};
+
+export const useTitle = (title?: string) => {
+  const { colony } = useColonyContext();
+  const { pathname } = useLocation();
+  const { formatMessage } = useIntl();
+  const { msg, values } = getMessageAndValues(pathname);
+
+  const colonyDisplayName = colony?.metadata?.displayName || colony?.name;
+
+  return useEffect(() => {
+    const titleToSet =
+      title ||
+      formatMessage(msg, {
+        ...values,
+        colonyName: colonyDisplayName,
+      });
+
+    document.title = titleToSet;
+  });
```

Comment:
Seeing that we don't really need this hook to return something, I think we can safely remove this `return`.